File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> cs;
        for(char c: s) {
            cs[c] += 1;
        }
        for (char c: t) {
            cs[c] -= 1;
        }
        for (const auto& [key, value] : cs)  {
            if (value != 0) {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-3.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 1;
        int right = piles[0]; 

        for (int i = 1; i < piles.length; i++) {
            if (piles[i] > right) {
                right = piles[i];
            }
        }
        System.out.println(right);
        while (left < right) {
            int mid = (left + right) / 2;
            System.out.println(mid);
            int speed = getSpeed(piles, mid);
            
            System.out.println(speed);
            if (speed > h) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return left;
    }
    public int getSpeed(int[] piles, int s) {
        int ans = 0;
        for (int i = 0; i < piles.length; i++) {
            ans += Math.ceil(((double) piles[i]) / ((double) s));
            System.out.println(ans);
        }
        return ans;
    }
}
